### PR TITLE
docs(Router): Update runGuardsAndResolvers documentation

### PR DESCRIPTION
**Pull request by ghost**
This is a PR for issue #46001

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/router/src/models.ts [hunk 1]
```diff
@@ -493,10 +495,14 @@ export interface Route {
 
   /**
    * Defines when guards and resolvers will be run. One of
-   * - `paramsOrQueryParamsChange` : Run when query parameters change.
    * - `always` : Run on every execution.
    * By default, guards and resolvers run only when the matrix
    * parameters of the route change.
+   * - `pathParamsChange` : Only re-run guards and resolvers when the path params change,
+   * ignores optional parameters and query params.
+   * - `paramsOrQueryParamsChange` : Run when query parameters change.
```
Comment (atscott) [inline packages/router/src/models.ts:503]:
```suggestion
   * - `paramsOrQueryParamsChange` : Run when path, matrix, or query parameters change.
```

File: packages/router/src/models.ts [hunk 2]
```diff
@@ -143,6 +143,8 @@ export type QueryParamsHandling = 'merge'|'preserve'|'';
 /**
  *
  * A policy for when to run guards and resolvers on a route.
+ * - `RunGuardsAndResolvers`: Run if the route is activated or deactivated and,
+ * the navigation doesn't change the activated route.
```
Comment (atscott) [inline packages/router/src/models.ts:147]: How about something more like: "Guards and resolvers will always run when a route is activated or deactivated. That is, when a navigation causes a change in the active `Route`. This option allows developers to control whether guards and resolvers run in cases where a route is unchanged. In this case, the default behavior behavior is to only run them when params change."

File: packages/router/src/models.ts [hunk 3]
```diff
@@ -493,10 +495,14 @@ export interface Route {
...
    * Defines when guards and resolvers will be run. One of
-   * - `paramsOrQueryParamsChange` : Run when query parameters change.
    * - `always` : Run on every execution.
    * By default, guards and resolvers run only when the matrix
    * parameters of the route change.
+   * - `pathParamsChange` : Only re-run guards and resolvers when the path params change,
+   * ignores optional parameters and query params.
+   * - `paramsOrQueryParamsChange` : Run when query parameters change.
+   * - `pathParamsOrQueryParamsChange` : Only re-run guards and resolvers when the path params change,
+   * also take the query params into consideration.
```
Comment (atscott) [inline packages/router/src/models.ts:505]:
```suggestion
   * - `pathParamsOrQueryParamsChange` : Re-run guards and resolvers when the path params change or
   * query params have changed. This does not include matrix parameters.
```

File: packages/router/src/models.ts [hunk 4]
```diff
@@ -493,10 +495,14 @@ export interface Route {
 
   /**
    * Defines when guards and resolvers will be run. One of
-   * - `paramsOrQueryParamsChange` : Run when query parameters change.
    * - `always` : Run on every execution.
    * By default, guards and resolvers run only when the matrix
    * parameters of the route change.
+   * - `pathParamsChange` : Only re-run guards and resolvers when the path params change,
+   * ignores optional parameters and query params.
```
Comment (atscott) [inline packages/router/src/models.ts:502]:
```suggestion
   * - `pathParamsChange` : Re-run guards and resolvers when the path params change. This does not compare matrix or query parameters.
```

**Review by atscott — CHANGES_REQUESTED** [review]
Hi, please also update the commit message to be 

"docs(Router): Update runGuardsAndResolvers documentation

Update documentation for runGuardsAndResolvers to be more clear about the conditions.

resolves #46001"

File: packages/router/src/models.ts (hunk 2 above)
Comment (najmiehsa) [reply packages/router/src/models.ts:147]: Hi, here on the current documentation the default case which is `paramsChange` is not mentioned. I suggest I include this above the `always` with the further description of "Re-run the guards and resolvers when path or path param change, this does not include query parameters." 
 
I can change the description for `RunGuardsAndResolvers` to  "Always run when a route is activated or deactivated and, a navigation causes a change in the active `route`.  In the case where a route is unchanged, the default behavior is the same as `paramsChange`.   Is this good?

Comment (atscott) [reply packages/router/src/models.ts:147]: Yea, that sounds good. Let's tweak the wording a bit though:

"Guards and resolvers will always run when a route is activated or deactivated or when a navigation causes a change in the active route. In the case where a route is unchanged, the default behavior is the same as `paramsChange`."

And "re-run" should be "rerun" without the dash.

**Review by atscott — CHANGES_REQUESTED** [review]
There are some other things not related to the documentation that need to be updated:

* The CLA needs to be signed
* The commits should be squashed. There only needs to be one commit for this
* The formatting lint check needs to be resolved.

Thanks!

**Comment by najmiehsa** [comment]
@atscott Thank you for the review. 
I've already signed the CLI yet it does not pass. This might be because I signed it with a Gmail but my git is with another email. 
I squashed all my commits into one single commit. 
I run the command for correct formatting of documentation but I'd get ``ng-dev `` doesn't exist. I tried to install developer packages but I am still in the process of it. Is there any other way to clean up the formatting here?

**Comment by atscott** [comment]
> I've already signed the CLI yet it does not pass. This might be because I signed it with a Gmail but my git is with another email.

That's right, you'll have to sign it with the email linked to your git account.

> I squashed all my commits into one single commit.

It looks like the commit message needs to be updated to conform to the repository requirements. It should be `docs(router)` rather than `docs(Router)`

> I run the command for correct formatting of documentation but I'd get `ng-dev ` doesn't exist. I tried to install developer packages but I am still in the process of it. Is there any other way to clean up the formatting here?

It uses clang-format so you could use clang format directly to format the files.

**Comment by najmiehsa** [comment]
Right now I see that CLA  and pullapprove are failing. Lint is passing so, I guess the format is alright. 
There is no option to edit and replace my initial email on the agreement. How can I get the CLA pass this?

**Comment by atscott** [comment]
@najmiehsa You can create a close this and open a new PR instead if you can't resolve the email issue.

**Pull request closed without merging**